runner/src/mill/runner/MillMain.scala

```diff
@@ -435,14 +438,37 @@ object MillMain {
     !(whitelistMatch || extraPlugins)
   }
 
+  def readVersionFile(file: os.Path): Option[String] = file match {
+    case f if os.exists(f) => os.read.lines(f).find(l => l.trim().nonEmpty)
+    case _ => None
+  }
+
+  private val usingMillVersionPattern = "//[|] +mill-version: +([^ ]+) *$".r
```

Comment:
We need to take care of quoting and yaml comments here.

Reply:
I'm not sure we should; in the end the lowest common denominator is the bash/bat parser, and that one can't really handle much beyond simple regex space-splitting. 

If we try to best effort handle as much of YAML as possible, it will end up being some arbitrary subset, since there's no way we can handle the whole of YAML in a bash regex. It might be better to just throw in the towel and spec that the `mill-version` key-value pair must have this specific syntax, at least that way we can clearly define what syntax is allowed (i.e. not much)

Reply:
I guess the question is if two-completely-separate-syntaxes is easier to understand, or if yaml-but-with-constraints is easier to understand. I'm guessing that 99% of people who write YAML will fall into writing `mill-version` "raw" and un-quoted, so although in theory it is a bit awkward in practice I think it'll be reasonably seamless for most people.

We could add support for simple quotes and comments as you mention, I think that actually won't be too hard even in Bash and Cmd. That should cover 99.9% of users since nobody is going to be using e.g. YAML multi-line strings for this even if in theory they could